File: apple_tree.rb
class NoApplesError < StandardError
  # This is an example of how we create custom exception classes
end

class AppleTree
  def age!
    # This method will age the tree by one year
  end

  def any_apples?
    # If any apples are present, true. If no apples, false
  end

  def pick_an_apple!
    raise NoApplesError, "This tree has no apples" unless self.any_apples?
    # we should return an apple if there are any apples. if there are no apples, raise the NoApplesError
  end
end

class Apple
  def initialize(diameter)
    # all apples have diameters, yeah? some are large and others are small
  end
end

File: apple_tree.py
from apple import Apple


class AppleTree:
    def __init__(self):
        self.age = 0
        self.height = 0
        self.apples = []

    def age_tree(self):
        self.age += 1
        if self.height < 99:
            self.height += 2

        for i in range(self.height // 2):
            self.apples.append(Apple())

    def is_dead(self):
        if self.age >= 100:
            return True
        else:
            return False

    def any_apples(self):
        if self.apples:
            return True
        else:
            return False

    def pick_an_apple(self):
        if self.apples:
            picked_apple = self.apples.pop()
            return picked_apple
        else:
            raise Exception('No apples on your tree')
